src/test/java/GmailServletTest.java

```diff
@@ -101,22 +166,277 @@ public void setUp() throws IOException, GeneralSecurityException {
+    printWriter.flush();
+    GmailResponse gmailResponse = gson.fromJson(stringWriter.toString(), GmailResponse.class);
+    Assert.assertEquals(0, gmailResponse.getUnreadEmailsHours());
+  }
+
+  @Test
+  public void checkDefaultUnreadImportantEmails() throws IOException, ServletException {
+    // For all queries, return no messages. Unread important emails should have 0 length
+    Mockito.when(request.getParameter(Mockito.eq("nDays")))
+        .thenReturn(String.valueOf(DEFAULT_N_DAYS));
+    Mockito.when(request.getParameter(Mockito.eq("mHours")))
+        .thenReturn(String.valueOf(DEFAULT_M_HOURS));
+
+    Mockito.when(gmailClient.listUserMessages(Mockito.anyString())).thenReturn(NO_MESSAGES);
+
+    servlet.doGet(request, response);
+    printWriter.flush();
+    GmailResponse gmailResponse = gson.fromJson(stringWriter.toString(), GmailResponse.class);
+    Assert.assertEquals(0, gmailResponse.getUnreadImportantEmails());
+  }
+
+  @Test
+  public void checkDefaultSender() throws IOException, ServletException {
+    Mockito.when(request.getParameter(Mockito.eq("nDays")))
+        .thenReturn(String.valueOf(DEFAULT_N_DAYS));
+    Mockito.when(request.getParameter(Mockito.eq("mHours")))
+        .thenReturn(String.valueOf(DEFAULT_M_HOURS));
+
+    Mockito.when(gmailClient.listUserMessages(Mockito.anyString())).thenReturn(NO_MESSAGES);
+
+    servlet.doGet(request, response);
+    printWriter.flush();
+    GmailResponse gmailResponse = gson.fromJson(stringWriter.toString(), GmailResponse.class);
+    Assert.assertEquals(DEFAULT_SENDER, gmailResponse.getSender());
+  }
+
+  @Test
+  public void checkSomeUnreadEmailsDays() throws IOException, ServletException {
+    Mockito.when(request.getParameter(Mockito.eq("nDays")))
+        .thenReturn(String.valueOf(DEFAULT_N_DAYS));
+    Mockito.when(request.getParameter(Mockito.eq("mHours")))
+        .thenReturn(String.valueOf(DEFAULT_M_HOURS));
+
+    Mockito.when(gmailClient.listUserMessages(Mockito.eq(UNREAD_EMAIL_DAYS_QUERY)))
+        .thenReturn(THREE_MESSAGES_MAJORITY_SENDER_WITH_NAME);
+    Mockito.when(gmailClient.listUserMessages(Mockito.eq(UNREAD_EMAIL_HOURS_QUERY)))
+        .thenReturn(NO_MESSAGES);
+    Mockito.when(gmailClient.listUserMessages(Mockito.eq(IMPORTANT_QUERY))).thenReturn(NO_MESSAGES);
+
+    Mockito.when(gmailClient.getUserMessage(Mockito.contains(MESSAGE_ID_ONE), Mockito.any()))
+        .thenReturn(THREE_MESSAGES_MAJORITY_SENDER_WITH_NAME.get(0));
+    Mockito.when(gmailClient.getUserMessage(Mockito.contains(MESSAGE_ID_TWO), Mockito.any()))
+        .thenReturn(THREE_MESSAGES_MAJORITY_SENDER_WITH_NAME.get(1));
+    Mockito.when(gmailClient.getUserMessage(Mockito.contains(MESSAGE_ID_THREE), Mockito.any()))
+        .thenReturn(THREE_MESSAGES_MAJORITY_SENDER_WITH_NAME.get(2));
+
+    servlet.doGet(request, response);
+    printWriter.flush();
+    GmailResponse gmailResponse = gson.fromJson(stringWriter.toString(), GmailResponse.class);
```

Comment:
I think an auxiliary method for these calls would help

Reply:
Agreed - fixed!